opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -418,52 +409,24 @@ def get_meter(
     This function is a convenience wrapper for
     opentelemetry.metrics.meter_provider().get_meter
     """
-    return meter_provider().get_meter(
+    return get_meter_provider().get_meter(
         instrumenting_module_name, stateful, instrumenting_library_version
     )
 
 
-def meter_provider() -> MeterProvider:
-    """Gets the current global :class:`~.MeterProvider` object.
+def set_meter_provider(meter_provider: MeterProvider) -> None:
+    """Sets the current global :class:`~.MeterProvider` object."""
+    global _METER_PROVIDER  # pylint: disable=global-statement
+    _METER_PROVIDER = meter_provider
 
-    If there isn't one set yet, a default will be loaded.
-    """
-    global _METER_PROVIDER, _METER_PROVIDER_FACTORY  # pylint:disable=global-statement
 
-    if _METER_PROVIDER is None:
-        # pylint:disable=protected-access
-        try:
-            _METER_PROVIDER = loader._load_impl(
-                MeterProvider, _METER_PROVIDER_FACTORY  # type: ignore
-            )
-        except TypeError:
-            # if we raised an exception trying to instantiate an
-            # abstract class, default to no-op meter impl
-            logger.warning(
-                "Unable to instantiate MeterProvider from meter provider factory.",
-                exc_info=True,
-            )
-            _METER_PROVIDER = DefaultMeterProvider()
-        _METER_PROVIDER_FACTORY = None
-
-    return _METER_PROVIDER
-
-
-def set_preferred_meter_provider_implementation(
-    factory: ImplementationFactory,
-) -> None:
-    """Set the factory to be used to create the meter provider.
-
-    See :mod:`opentelemetry.util.loader` for details.
-
-    This function may not be called after a meter is already loaded.
-
-    Args:
-        factory: Callback that should create a new :class:`MeterProvider` instance.
-    """
-    global _METER_PROVIDER_FACTORY  # pylint:disable=global-statement
+def get_meter_provider() -> MeterProvider:
+    """Gets the current global :class:`~.MeterProvider` object."""
+    global _METER_PROVIDER  # pylint: disable=global-statement
 
-    if _METER_PROVIDER:
-        raise RuntimeError("MeterProvider already loaded.")
+    if _METER_PROVIDER is None:
+        _METER_PROVIDER = (
+            Configuration().meter_provider  # type: ignore # pylint: disable=no-member
```

Comment:
should Configuration also be a lazily evaluated global as well? I guess I wonder if we need to re-initialize the class every single time a new one is created.

Also using the global pattern would allow one to directly set configuration, if they wanted to. I personally prefer a more programmatic control of application configuration.

Reply:
The access of the `meter_provider` or `tracer_provider` attributes is lazy. The `Configuration` object is a singleton.

An important part of this configuration manager is to separate code from configuration, would setting directly the configuration in code be against that?